week1/homework/src/server.js

```diff
@@ -1,20 +1,48 @@
+
 'use strict';
 
 const http = require('http');
 
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
- */
+function printData(state, response) {
+  response.writeHead(200, {
+    'Content-Type': 'application/json'
+  });
+  let json =
+  response.write(JSON.stringify({
+    'state': state
+  }));
+}
+
 function createServer(port) {
   let state = 10;
 
-  const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+  const server = http.createServer((req, response) => {
+    if (req.url == '/state') {
+      printData(state, response);
+    }
+    else if (req.url == '/add') {
+      state += 1;
+      printData(state, response);
+    }
+    else if (req.url == '/subtract') {
+      state -= 1;
+      printData(state, response);
+    }
+    else if (req.url == '/reset') {
+      state = 10;
+      printData(state, response);
+    }
+    else {
+      response.writeHead(404, {
+        'Content-Type': 'application/json'});
```

Comment:
This is the same thing you do on line 7 och  8, maybe you could move this in to you `printData` function ?